### PR TITLE
Fix Days.daysBetween int overflow

### PR DESCRIPTION
**Pull request by TuanNguyen27**
`Days.daysBetween(datetime_1, datetime_2)` could throw `java.lang.ArithmeticException: Value cannot fit in an int:` if the duration between two dates in milliseconds cannot fit a 32-bit int representation. See related Stack Overflow issue [here](https://stackoverflow.com/questions/25204043/joda-time-calculation-of-seconds-between-two-dates-throws-an-exception)

**Describe the proposed solution**
Switch to using joda's `Duration`. 

**Describe alternatives you've considered**
N/A 

**Additional context**
N/A

### CODE REVIEW

File: core/src/test/scala/com/salesforce/op/stages/impl/feature/DateMapVectorizerTest.scala [hunk 1]
```diff
@@ -122,7 +122,7 @@ class DateMapVectorizerTest extends FlatSpec with TestSparkContext with Attribut
   }
 
   it should "vectorize dates correctly on test date" in {
-    checkAt(new JDateTime(2017, 9, 28, 15, 45, 39, DateTimeUtils.DefaultTimeZone))
+    checkAt(new JDateTime(1901, 1, 1, 0, 0, 0, DateTimeUtils.DefaultTimeZone))
```
Comment (tovbinm) [inline core/src/test/scala/com/salesforce/op/stages/impl/feature/DateMapVectorizerTest.scala:125]: better add a new test case

File: core/src/main/scala/com/salesforce/op/stages/impl/feature/OPMapVectorizer.scala [hunk 2]
```diff
@@ -397,7 +397,7 @@ final class DateMapVectorizerModel[T <: OPMap[Long]] private[op]
   val timeZone: DateTimeZone = DateTimeUtils.DefaultTimeZone
 
   def convertFn: DateMap#Value => RealMap#Value = (dt: DateMap#Value) =>
-    dt.mapValues(v => Days.daysBetween(new DateTime(v, timeZone), referenceDate).getDays.toDouble)
+    dt.mapValues(v => new Duration(new DateTime(v, timeZone), referenceDate).getStandardDays.toDouble)
```
Comment (tovbinm) [inline core/src/main/scala/com/salesforce/op/stages/impl/feature/OPMapVectorizer.scala:400]: also, there are more places like this in the code. try grepping.

Comment (TuanNguyen27) [reply core/src/main/scala/com/salesforce/op/stages/impl/feature/OPMapVectorizer.scala:400]: done!

Comment (gerashegalov) [reply core/src/main/scala/com/salesforce/op/stages/impl/feature/OPMapVectorizer.scala:400]: there is at least one more in `DateTimeVectorizerTest`

File: core/src/test/scala/com/salesforce/op/stages/impl/feature/DateMapVectorizerTest.scala (hunk 1 above)
Comment (gerashegalov) [reply core/src/test/scala/com/salesforce/op/stages/impl/feature/DateMapVectorizerTest.scala:127]: @TuanNguyen27 please add test cases for ArithmeticExceptions you  are trying to solve

**Comment by gerashegalov** [comment]
please add a real PR description

File: core/src/test/scala/com/salesforce/op/stages/impl/feature/DateMapVectorizerTest.scala (hunk 1 above)
Comment (TuanNguyen27) [reply core/src/test/scala/com/salesforce/op/stages/impl/feature/DateMapVectorizerTest.scala:127]: @gerashegalov the new test case is `checkAt(new JDateTime(1901, 1, 1, 0, 0, 0, DateTimeUtils.DefaultTimeZone))`, which exceed the limit if we use `Days.daysBetween`. Does that work ?

**Pull request merged**